githooks/prepare-commit-message.sample

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+#
+# An example hook script to prepare the commit log message.
+# Called by "git commit" with the name of the file that has the
+# commit message, followed by the description of the commit
+# message's source.  The hook's purpose is to edit the commit
+# message file.  If the hook fails with a non-zero status,
+# the commit is aborted.
+#
+# To enable this hook, rename this file to "prepare-commit-msg".
+#
+# Automatically adds Topic-Placeholders, User, JIRA-Ticket and branch description (if available) to every commit message.
+#
+
+JIRA_TICKET=$(git rev-parse --abbrev-ref HEAD | grep -Eo '([A-Z]{3,}-)([0-9]+)' -m 1)
+USER=git var GIT_AUTHOR_IDENT
+NAME=$(git branch | grep '*' | sed 's/* //') 
+DESCRIPTION=$(git config branch."$NAME".description)
+
+echo "[<TOPIC> : $USER] <DETAILS> - $JIRA_TICKET"|cat - "$1" > /tmp/out && mv /tmp/out "$1"
```

Comment:
Providing this formatting may be a feature we need to set up as configurable. Maybe we can find some commonly used items like `$USER`, `<TOPIC>`, and `<DETAILS>` and set up some way for the user to pick and choose how they want to set up the placeholder.

What I wanted was to just have 

```
"$JIRA_TICKET - "
```
As we keep our commit messages pretty simple. 


Some advice I got from another dev here was to set have my shell scripts check for certain environment variables before running. If they aren't set, we could force the user to set them before we could continue. One variable could be `$GIT_HOOK_PLACEHOLDER`, which may somehow include any template text the user my want.

Possible example usage:
```
$GIT_HOOK_PLACEHOLDER="$JIRA_TICKET - "

# or

$GIT_HOOK_PLACEHOLDER="[<TOPIC> : $USER] <DETAILS> - $JIRA_TICKET"
```

This would then change line 20 to:

```
echo "$GIT_HOOK_PLACEHOLDER | cat - "$1" > /tmp/out && mv /tmp/out "$1"
```

Reply:
We would somehow need to check for the $GIT_HOOK_PLACEHOLDER env var. if it isn't set, i dont think we should exit 1, but probably just not do anything at all and continue with the default commit hook template.